metrics/sample_test.go

```diff
@@ -235,6 +146,8 @@ func TestResolversForTrendColumnsCalculation(t *testing.T) {
 	for _, tc := range customResolversTests {
 		tc := tc
 		t.Run(fmt.Sprintf("%v", tc.stats), func(t *testing.T) {
+			t.Parallel()
+
```

Comment:
The race from https://github.com/grafana/k6/pull/2426#issuecomment-1064860615 are from this . On line 155 it is in parallel touching the sink that is made just before the `for` . Moving the creation of the `sink` just after the `t.Parallel()` seems to fix it .